src/main/Makefile.am

```diff
@@ -28,10 +28,11 @@ pcvc4_SOURCES = \
 	driver_unified.cpp
 pcvc4_LDADD = \
 	libmain.a \
-	@builddir@/../parser/libcvc4parser.la \
-	@builddir@/../libcvc4.la \
 	$(READLINE_LIBS) \
-	@builddir@/../lib/libreplacements.la
+	@builddir@/../lib/libreplacements.la \
+	@builddir@/../libcvc4.la \
+	@builddir@/../parser/libcvc4parser.la \
+	@builddir@/../util/libutil.la
```

Comment:
So I figured out the issue. There are two historically "correct" fixes for this in the CVC4 project. But including libutil here is neither of them. (You can disagree with this, but this is the history.) The problem lies in what symbols are exported in libcvc4.so . This also involves the distinction between cvc4_private.h , cvc4_private_library.h and cvc4_public.h .

Steps to reproduce the bug:
$ git clone https://github.com/4tXJ7f/CVC4.git andres
$ cd andres/
$ git checkout signal 
$ git checkout master -- src/main/Makefile.am
$ ./autogen.sh ; ./configure --with-build=production --disable-proof --with-antlr-dir=/home/taking/install_targets/antlr/antlr3.4/ ANTLR=/home/taking/install_targets/antlr/antlr3.4/bin/antlr3 ; make -j 5

The issue is revealed by the following two commands:
$ nm -C builds/src/util/.libs/libutil.a | grep safe_print
safe_print.o:
0000000000000000 t _GLOBAL__sub_I_safe_print.cpp
0000000000000670 T void CVC4::safe_print<timespec>(int, timespec)
[more stuff]
$ nm -C builds/src/.libs/libcvc4.so | grep safe_print
00000000000b91b0 t _GLOBAL__sub_I_safe_print.cpp
00000000008b2150 t void CVC4::safe_print<timespec>(int, timespec)
[more stuff]

The problem is the "t" vs. "T" there. That probably makes absolutely no sense. (It did not to me a when I first encountered this.) Luckily the intro to this rabbit hole is documented here:
https://github.com/CVC4/CVC4/blob/master/src/util/statistics_registry.h#L18
Make sure to read the linked article to understand why those commands demonstrate what the problem is. You also probably want to read the headers:
https://github.com/CVC4/CVC4/blob/master/src/include/cvc4_private.h
https://github.com/CVC4/CVC4/blob/master/src/include/cvc4_public.h
https://github.com/CVC4/CVC4/blob/master/src/include/cvc4_private_library.h
to understand what CVC4_PUBLIC is and what problem cvc4_private_library.h is solving. Once you understand what the problem is, it takes about 2 mins to implement one of those two solutions. We can then debate over which of the two solutions is better :-)

Ping me if you spend more than a couple of hours on this.